src/components/navigation/utils.ts

```diff
@@ -0,0 +1,5 @@
+import { ComponentNode, TreeNode } from './types';
+
+export default function isComponentNode(tree: TreeNode): tree is ComponentNode {
+  return 'component' in tree;
```

Comment:
I'd love to use `tree.hasOwnProperty('component')`, but eslint didn't like that. But I guess there's no much difference in such a straightforward case.

Reply:
Either solutions are fine to me :) 
Would you mind converting this into an expression with an anonymous function to keep consistency with the rest of the code? `const isComponentNode = (tree: TreeNode): tree is ComponentNode => {....}`